app/lib/boolean-phrase-parser.js

```diff
@@ -0,0 +1,208 @@
+(function (_) {
```

Comment:
Maybe we can eliminate the dependency on EJS altogether because it is being passed in the IIFE that surrounds the entire lib (see header.jsp)